substrate/frame/support/src/genesis_builder_helper.rs

```diff
@@ -55,3 +55,82 @@ where
 		preset_for_name,
 	)
 }
+
+/// Creates a `RuntimeGenesisConfig` JSON patch.
+///
+/// This macro creates a default `RuntimeGenesisConfig` initializing provided fields with given
+/// values, serialize it to JSON blob, and retain only the specified fields.
+///
+/// This macro helps to prevents errors that could occur from manually creating JSON objects, such
+/// as typos or discrepancies caused by future changes to the `RuntimeGenesisConfig` structure. By
+/// using the actual struct, it ensures that the JSON generated is valid and up-to-date.
+///
+/// This macro assumes that `serde(rename_all = "camelCase")` attribute is used for
+/// RuntimeGenesisConfig, what should be the case for frame-based runtimes.
+///
+/// # Example
+///
+/// ```rust
+/// use frame_support::runtime_genesis_config_json;
+/// #[derive(Default, serde::Serialize, serde::Deserialize)]
+/// #[serde(rename_all = "camelCase")]
+/// struct RuntimeGenesisConfig {
+///     a_field: u32,
+///     b_field: u32,
+///     c_field: u32,
+/// }
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31, b_field:127}),
+///     serde_json::json!({"aField":31, "bField":127})
+/// );
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31}),
+///     serde_json::json!({"aField":31})
+/// );
+/// ```
+#[macro_export]
+macro_rules! runtime_genesis_config_json {
```

Comment:
Not sure if this is needed, but we could allow to optionally specify the name of the struct to use. Such that one could to both `genesis_config_json!({a_field: 31, b_field: 127})` and `genesis_config_json!(CustomGenesisConfig, {x_field: 31, y_field: 127})`. Not sure if needed, but would make it more flexible. But have no strong feelings about it.

Reply:
Or if you manually want to use a different name for whatever reason.  e.g. use `crate::GenesisRuntimeConfig as CustomConfig`. But yeah maybe too niche. If someone wants this, we could add later.

Reply:
done in: 6db4b41